### PR TITLE
server/snap: make transport configurable

### PR DESCRIPTION
**Pull request by BusyJay**
Different cluster may require a different value for its scale and performance.

### CODE REVIEW

**Comment by huachaohuang** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Comment by BusyJay** [comment]
PTAL

File: src/server/snap.rs [hunk 1]
```diff
@@ -154,8 +151,10 @@ fn send_snap(
     };
 
     let cb = ChannelBuilder::new(env)
```
Comment (ngaut) [inline src/server/snap.rs]: Why gRPC related things changed?

File: etc/config-template.toml [hunk 2]
```diff
@@ -61,6 +61,11 @@
 # Amount to read ahead on individual grpc streams.
 # grpc-stream-initial-window-size = "2MB"
 
+# How many snapshots can be sent concurrently.
+# concurrent-send-snap-limit = 32
```
Comment (ngaut) [inline etc/config-template.toml]: Is the default value too high?

File: src/server/snap.rs (hunk 1 above)
Comment (BusyJay) [reply src/server/snap.rs]: These configurations are used to optimize the transport of snapshot just like send/recv limit.

File: etc/config-template.toml (hunk 2 above)
Comment (BusyJay) [reply etc/config-template.toml]: No. For most server's configuration, 32 is OK.

**Comment by hicqu** [comment]
LGTM.

File: src/server/snap.rs [hunk 3]
```diff
@@ -154,8 +151,10 @@ fn send_snap(
...
-        .keepalive_time(Duration::from_secs(60))
-        .keepalive_timeout(Duration::from_secs(3));
+        .stream_initial_window_size(cfg.grpc_stream_initial_window_size.0 as usize)
```
Comment (overvenus) [inline src/server/snap.rs]: It may overflow.

Comment (BusyJay) [reply src/server/snap.rs]: Yes. Just like all other similar places in the codebase, it should be fixed after grpcio is updated.

**Pull request merged**